polygeist/tools/cgeist/Test/Verification/compoundassign.c

```diff
@@ -171,3 +173,64 @@ unsigned f11(unsigned *a, unsigned b) {
 unsigned f12(unsigned *a, unsigned b) {
   return *a ^= b;
 }
+
+// COM: This test is incorrect due to signedness detection issues.
+
+// CHECK-LABEL:   func.func @f13(
+// CHECK-SAME:                   %[[VAL_0:.*]]: memref<?xi8>,
+// CHECK-SAME:                   %[[VAL_1:.*]]: i32) -> i8
+// CHECK:           %[[VAL_2:.*]] = arith.constant 0 : i32
+// CHECK:           %[[VAL_3:.*]] = affine.load %[[VAL_0]][0] : memref<?xi8>
+// CHECK:           %[[VAL_4:.*]] = arith.extui %[[VAL_3]] : i8 to i32
+// CHECK:           %[[VAL_5:.*]] = arith.addi %[[VAL_4]], %[[VAL_1]] : i32
+// CHECK:           %[[VAL_6:.*]] = arith.cmpi ne, %[[VAL_5]], %[[VAL_2]] : i32
+// CHECK:           %[[VAL_7:.*]] = arith.extui %[[VAL_6]] : i1 to i8
+// CHECK:           affine.store %[[VAL_7]], %[[VAL_0]][0] : memref<?xi8>
+// CHECK:           return %[[VAL_7]] : i8
+// CHECK:         }
+
+bool f13(bool *a, unsigned b) {
+  return *a += b;
+}
+
+// COM: Currently not working for how +operator is implemented for pointers.
+
+#if 0
+bool f14(bool *a, void *b) {
+  return *a += b;
+}
+#endif
+
+// CHECK-LABEL:   func.func @f15(
+// CHECK-SAME:                   %[[VAL_0:.*]]: memref<?xi8>,
+// CHECK-SAME:                   %[[VAL_1:.*]]: f32) -> i8
```

Comment:
This is incorrect because the function should return an i1 not an i8.
Note: the LLVM IR generated by clang for this function signature is: 
```
define dso_local zeroext i1 @_Z3f16Pbd(i8* %0, double %1) #0 {

Reply:
You're right, I'm aware of this problem. I should've mentioned this PR is intended to be part of a series of PRs addressing #7314. I'll add a list of tasks to be performed to support `i1` in that issue for clarification.

Reply:
Excellent, good to create issues for things that we find are incorrect (and there are many in cgeist unfort).